### DISCUSSION TITLE
Lighthouse deck error (The deck bitstream does not match the required bitstream)

### DISCUSSION THREAD

**Discussion by fehmi8** [bitcraze/discussions#96] [Q&A] [answered]
Hi there,

We encountered the following issue with the lighthouse decks, after flashing to the latest firmware. 
Both drone and lighthouse deck are brand new, out of the box. Changing the lighthouse deck solves the problem. Out of 22 lighthouse decks, 8 of them show a red light with the following error:  

```
SYS: ----------------------------
SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:8c606d1f8f89 (2022.05) CLEAN
SYS: I am 0x203934305333500600490032 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcLighthouse4 for deck 0
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2022.0
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 19240 bytes
LHFL: Lighthouse bootloader version: 2
STAB: Ready to fly.
LHFL: Bitstream CRC32: 952E7A01 [FAIL]
LHFL: The deck bitstream does not match the required bitstream.
LHFL: We require lighthouse bitstream of size 104093 and CRC32 112BC794.
LHFL: Leaving the deck in bootloader mode ...
LH: FPGA not booted. Lighthouse disabled!
```

The following post described similar issues, but the solution didn't work:
https://forum.bitcraze.io/viewtopic.php?t=4763

Any idea how to solve this? Thanks for your help!

**Comment by krichardsson** [comment]
Hi!

Deck binaries are updated in the flashing process when you flash a full release from the client. All you have to do is to mount the deck on your crazyflie and do a [firmware upgrade](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#firmware-upgrade) from the client.

**Comment by fehmi8** [comment]
Hej Kristoffer!

Thanks for the quick answer! Indeed the core problem is that the deck doesn't get upgraded. When launching the upgrade (2022.05), the cf2 firmware gets upgraded but after rebooting,  it shows "identifying deck to be updated" and then goes to IDLE, without updating the lighthouse deck.

Sorry for the bad quality video (edited to be shorter):
https://youtu.be/fi9EKvtoC7E

Any clue?

**Reply by ralphtoonsutd** [reply]
Hi there,

I was experiencing this problem with new purchased lighthouse decks recently, and found that having another deck attached to the crazyflie was blocking the lighthouse deck updating. When I attached only the lighthouse deck to the crazyflie it seemed to update okay (most recent client and stock firmware used). Hope this helps!

**Reply by knmcguire** [reply]
It seems that @ralphtoonsutd is right! The same thing also happens with the AIdeck in combination any other deck. The lighthouse deck should be the only deck attached and in the video I see that you also attached the multiranger deck. This is indeed an interesting find and I guess kind of a bug (or at least should have been a warning). I made this issue for it in the cfclient: https://github.com/bitcraze/crazyflie-clients-python/issues/616

@fehmi8 Could** you remove all other decks except for the lighthouse deck and try again?

**Comment by fehmi8** [comment, marked as answer]
> It seems that @ralphtoonsutd is right! The same thing also happens with the AIdeck in combination any other deck. The lighthouse deck should be the only deck attached and in the video I see that you also attached the multiranger deck. This is indeed an interesting find and I guess kind of a bug (or at least should have been a warning). I made this issue for it in the cfclient: [bitcraze/crazyflie-clients-python#616](https://github.com/bitcraze/crazyflie-clients-python/issues/616)
> 
> @fehmi8 Could** you remove all other decks except for the lighthouse deck and try again?

This worked, thanks Kimberly!

**Reply by knmcguire** [reply]
good it worked! and kudos to @ralphtoonsutd ofcourse
